src/shared/util/convex_hull.cpp

```diff
@@ -0,0 +1,87 @@
+#include "convex_hull.h"
+#include <iostream>
+
+using Points = std::vector<GVector::vector2d<int>>;
+
+struct {
+  bool operator()(const GVector::vector2d<int> &a, const GVector::vector2d<int> &b) const {
+    return (a.x < b.x) || (a.x == b.x && a.y < b.y);
+  }
+} lexicographicLess;
+
+/*
+  Implementation of Andrew's monotone chain convex hull algorithm (O(n log n))
+  Returns points in clockwise-order.
+*/
+Points computeConvexHullFromPoints(Points &P) {
+  const int n = P.size();
+  if (n == 1) return P;
+  
+  Points R(2 * n);
+  std::sort(P.begin(), P.end(), lexicographicLess);
+
+  int k = 0;
+  // Lower hull
```

Comment:
How many points are usually used? If it is large, I suggest using one of the O(n log(n)) algorithms like Graham Scan: https://en.wikipedia.org/wiki/Convex_hull_algorithms
Maybe this can be tracked as an enhancement feature for later optimization.

Reply:
depends on the camera distortion. With the new FLIR cameras, 4 points are already sufficient.
@rhololkeolke how many points have you used?

Reply:
It depends on how distorted your cameras are and how much noise you have around the field. I think we were using about 8 points for some of the cameras in our previous lab setup. 

I had put a version of this on the field computers at last years competition and I think it was between 4 and 8 points. I wasn't super exact with the boundaries. Just used it to cut out the crowds walking by and robots sitting on the tables.

Reply:
so it seems there is currently no need to optimize it. the convex hull is used to generate the masked image on changes only anyway, so this should not have an influence on the image processing time.